pkg/token/execCredentialWriter.go

```diff
@@ -3,37 +3,91 @@ package token
 //go:generate sh -c "mockgen -destination mock_$GOPACKAGE/execCredentialWriter.go github.com/Azure/kubelogin/pkg/token ExecCredentialWriter"
 
 import (
+	"bytes"
 	"encoding/json"
 	"fmt"
 	"os"
 
 	"github.com/Azure/go-autorest/autorest/adal"
-	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/pkg/apis/clientauthentication"
+	v1 "k8s.io/client-go/pkg/apis/clientauthentication/v1"
 	"k8s.io/client-go/pkg/apis/clientauthentication/v1beta1"
 )
 
+const (
+	apiV1       string = "client.authentication.k8s.io/v1"
+	apiV1beta1  string = "client.authentication.k8s.io/v1beta1"
+	execInfoEnv string = "KUBERNETES_EXEC_INFO"
+)
+
 type ExecCredentialWriter interface {
-	Write(token adal.Token) error
+	Write(token adal.Token, buffer bytes.Buffer) error
```

Comment:
```suggestion
	Write(token adal.Token, writer io.Writer) error
```
it should be like this so that you pass in os.Stdout from kubelogin and pass in buf.Buffer from unit test

Reply:
Done in the current commit. Thanks!